### ISSUE TITLE
Dark mode on beta dashboard

### ISSUE THREAD

**Issue by pllim** [readthedocs/ext-theme#216]
Would be nice if I can toggle dark mode on https://beta.readthedocs.org/dashboard/ . @humitos said I could submit this feature request here. Thank you for your consideration and for this wonderful project!

**Comment by humitos** [comment]
There are some history context about dark mode on the dashboard in https://github.com/readthedocs/readthedocs.org/issues/3819

**Comment by agjohnson** [comment]
I had actually experimented with this early on. This is very possible and is somewhere on our dashboard roadmap. We'll be taking on more new feature work later on in the year, so likely considering it then.

Just a note: the issue linked above is actually mostly folks wanting a dark mode on our Sphinx theme and documentation projects, not the dashboard -- it's difficult to communicate the difference there though.

- https://github.com/readthedocs/readthedocs.org/issues/3819

On the new dashboard, we're using FomanticUI (SemanticUI), which does not yet have a dark mode theme. This is a big project for FUI:

- https://github.com/fomantic/Fomantic-UI/issues/1424

What I did experiment with early on is switching the elements used in our themes, so a FUI `.ui.segment` would become a `.ui.inverted.segment` with a dark mode enabled. This is native FUI, which is nice, but does require some creative solutions in templates or maybe client side.

I was a big fan of the end result though. I think the best way to do this is:

- Add a user profile settings for dark mode with options: enabled, disable, and sync with system. I do not like sync with system as the default option though, this is usually a usability headache for users.
- Unknown right now, but I think we'd have to do the bulk of this in templates. Templates would have a template filter to give something like `<div class="ui {% dark_mode "inverted" %} segment %}"></div>`.
- We have to sprinkle `inverted` on many classes as it doesn't trickle up the DOM

Here's a quick demo of what this ends up looking like:

![image](https://github.com/user-attachments/assets/fb1823b1-c97a-4a4f-a3d2-4adf67f64fda)

**Comment by pllim** [comment]
FWIW default to light mode is probably least surprise to users.

I like that screenshot. Looking forward to it being in production. Thanks!

**Comment by flying-sheep** [comment]
Defaulting to [`@media/prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) is definitely the least surprise to users!

**Comment by agjohnson** [comment]
I was recently in the LESS docs and realized that the `extends()` function actually could be used to extend all of the inverted SUI CSS rules and, if it worked, could pretty easily get us a dark mode set of rules in our theme.

So I went through the rules this afternoon and it actually came together rather quickly. I have probably about 95% of the dashboard working with dark mode currently, most of the remaining bugs aren't super noticeable for the most part.

<img width="2029" height="1571" alt="Image" src="https://github.com/user-attachments/assets/444d5ec4-3ae1-460f-bec5-36d6b419dd0d" />

So dark mode could be coming very soon. I haven't decided the best way to roll this out, I do prefer when sites give a preference of `light, dark, system` though. So I'd probably want to aim for some version of this, but that could also be a next iteration thing.

- #727

**Comment by flying-sheep** [comment]
Things got much easier in recent years using vanilla CSS. You can now use `light-dark(color1, color2)` to switch colors based on the *current* color scheme, and override the system color scheme by e.g. via `document.documentElement.style.colorScheme = "dark"`

makes for an absolutely trivial theme system that can even be overridden in sub-elements without any problem

**Comment by agjohnson** [comment]
Absolutely, between that and CSS variables features like dark mode and theming are way easier to implement with CSS now.

In this case, we're using FomanticUI (SemanticUI) and don't want to stray too far into maintaining CSS outside the existing framework. So we are a bit limited with the structure there. While there are styles for both light and dark elements, these styles were only intended to be used explicitly -- the elements weren't intended to respond to shifts in light/dark themes automatically. CSS variable usage is a ways out there too, but there are discussions of both.

I'm happy this hack has worked out though, I'd say it's likely you might notice dark mode in one of the upcoming releases. And perhaps a later release after that might implement a light/dark/system theme selection dropdown somewhere.

**Comment by agjohnson** [comment]
With our release last week, dark mode is now live and you can enable it for your account. We'll have more docs and will get word out about this soon. For now, anyone that is interested in using or testing dark mode can now enable both dark mode or dark mode by system theme in your user preferences dashboard:

https://app.readthedocs.org/accounts/preferences/

<img width="1867" height="932" alt="Image" src="https://github.com/user-attachments/assets/acb904bf-3f9c-46a0-b7cc-a4d294e73c96" />

There are a few elements not yet styled and there could be some display issues lingering. There are a few bugs and future changes in:

- #747 

Feel free to note issues there or open up an issue for anything that stands out!
